modules/delegates/api/__tests__/filterDelegateAddresses.spec.ts

```diff
@@ -0,0 +1,203 @@
+/*
+
+SPDX-FileCopyrightText: © 2023 Dai Foundation <www.daifoundation.org>
+
+SPDX-License-Identifier: AGPL-3.0-or-later
+
+*/
+
+import { DelegateTypeEnum } from 'modules/delegates/delegates.constants';
+import { filterDelegateAddresses } from 'modules/delegates/helpers/filterDelegates';
+
+const delegates = [
+  {
+    blockTimestamp: new Date('2021-07-27T05:59:34.000Z'),
+    expirationDate: new Date('2022-07-27T05:59:34.000Z'),
+    isAboutToExpire: true,
+    delegate: '0xd52623ee9a40402a5a9ed82bb0417e04d88a778c',
+    voteDelegate: '0xb21e535fb349e4ef0520318acfe589e174b0126b',
+    expired: false,
+    name: 'coldiron.eth',
+    delegateType: DelegateTypeEnum.RECOGNIZED,
+    tags: ['growth', 'risk', 'security', 'decentralization'],
+    previous: undefined,
+    next: {
+      address: '0xe491d9ea59cae5dd84804b224535c64d5cb90c3a',
+      voteDelegateAddress: '0xb4b82978fce6d26a22dea7e653bb9ce8e14f8056'
+    }
+  },
```

Comment:
Maybe we can move this to a json file so it could be used in other places as well. For example, we have a mock polls json at `modules/polling/api/mocks`